<?xml version="1.0" ?>

<launch>
    <arg name="use_gui" default ="False" />
    <arg name="pybullet" default ="False" />
    <arg name="render_pybullet" default ="True" />
    <arg name="dt" default="0.1"/>
    <arg name="sensor_method" default="rgb"/>

    <arg name="fingerprint_names" default="ball block blank"/>
    <arg name="fp_base_name" default="_belief_[]_L2_LatentSpace_final.pickle"/>

    <arg name="cam_z" default="0.46"/>
    <arg name="save_folder" default="test_fingerprints/"/>
    <arg name="test_path" default="data/rgb/entklerg_0001_ddp_2obj_z16_newlights_trainMulti" doc="path relative to franka_test directory"/>
    <arg name="fingerprint_path" default="eval/" doc="path relative to test_path"/>
    <arg name="model_path" default="model_final.pth" doc="name of model to test"/>
    <arg name="explr_states" default="xy" doc="states to explore"/>
    <arg name="angle_method" default="mean" doc="angle reduction method (mean,max,Max2,etc.)"/>
    <arg name="type" default="LatentSpace" doc="saved heatmap type (LatentSpace,OutputError)"/>

    <!-- params -->
    <param name="base_path" value="$(find franka_test)" type="string" />
    <param name="pybullet" value="$(arg pybullet)" type="bool" />
    <param name="test_path" value="$(arg test_path)" type="string" />
    <param name="dt" value="$(arg dt)" type="double" />
    <param name="save_folder" value="$(arg save_folder)" type="string" />
    <param name="fingerprint_names" value="$(arg fingerprint_names)" type="string" />
    <param name="render_pybullet" value="$(arg render_pybullet)" type="bool" />
    <param name="cam_z" value="$(arg cam_z)" type="double" />
    <param name="model_path" value="$(arg model_path)" type="string" />
    <param name="explr_states" value="$(arg explr_states)" type="string" />
    <param name="fingerprint_path" value="$(arg fingerprint_path)" type="string" />
    <param name="fp_base_name" value="$(arg fp_base_name)" type="string" />
    <param name="angle_method" value="$(arg angle_method)" type="string" />
    <param name="type" value="$(arg type)" type="string" />
    <param name="sensor_method" value="$(arg sensor_method)" type="string" />

    <!-- test nodes -->
    <node name="capture_fingerprint_belief" pkg="franka_test" type="capture_fingerprint_belief" respawn="false" output="screen" required="true"/>

    <!-- image nodes -->
    <group if="$(eval (arg('sensor_method')=='rgb') or (arg('sensor_method')=='intensity'))">
      <arg name="img_name" value="/usb_cam/image_raw" />

      <node name="republish" type="republish" pkg="image_transport" output="screen" args="compressed in:=$(arg img_name) raw out:=/usb_cam/image" unless="$(arg pybullet)"/>

      <!-- <node name="republish" type="republish" pkg="image_transport" output="screen" args="compressed in:=$(arg img_name) raw out:=/usb_cam/image_raw" if="$(arg pybullet)"/> -->
    </group>

    <!-- <node name="pybullet_service" pkg="franka_test" type="pybullet_service" respawn="false" output="log" if="$(arg pybullet)" /> -->

    <node name="gui" pkg="franka_test" type="gui" respawn="false" output="screen" if="$(arg use_gui)" />

</launch>
